android/src/main/java/com/reactcommunity/rndatetimepicker/RNMaterialRangePicker.kt

```diff
@@ -0,0 +1,207 @@
+      builder.setInputMode(MaterialDatePicker.INPUT_MODE_CALENDAR)
+    }
+  }
+
+  private fun setConstraints() {
+    val constraintsBuilder = CalendarConstraints.Builder()
+
+    if (args.containsKey(RNConstants.FIRST_DAY_OF_WEEK)) {
+      constraintsBuilder.setFirstDayOfWeek(args.getInt(RNConstants.FIRST_DAY_OF_WEEK))
+    }
+
+    val validators = mutableListOf<DateValidator>()
+
+    if (args.containsKey(RNConstants.ARG_MINDATE)) {
+      val minDate = Common.minDateWithTimeZone(args)
+      validators.add(DateValidatorPointForward.from(minDate))
+    }
+
+    if (args.containsKey(RNConstants.ARG_MAXDATE)) {
+      val maxDate = Common.maxDateWithTimeZone(args)
+      validators.add(DateValidatorPointBackward.before(maxDate))
+    }
+
+    constraintsBuilder.setValidator(CompositeDateValidator.allOf(validators))
+    builder.setCalendarConstraints(constraintsBuilder.build())
+  }
+
+  private fun setFullscreen() {
+    val isFullscreen = args.getBoolean(RNConstants.ARG_FULLSCREEN)
+
+    if (isFullscreen) {
+      builder.setTheme(com.google.android.material.R.style.ThemeOverlay_Material3_MaterialCalendar_Fullscreen)
+    } else {
+      builder.setTheme(com.google.android.material.R.style.ThemeOverlay_Material3_MaterialCalendar)
+    }
+  }
+
+  private fun setButtons() {
+    val buttons = args.getBundle(RNConstants.ARG_DIALOG_BUTTONS) ?: return
+
+    val negativeButton = buttons.getBundle(Common.NEGATIVE)
+    val positiveButton = buttons.getBundle(Common.POSITIVE)
+
+    if (negativeButton != null) {
+      builder.setNegativeButtonText(negativeButton.getString(Common.LABEL))
+    }
+
+    if (positiveButton != null) {
+      builder.setPositiveButtonText(positiveButton.getString(Common.LABEL))
+    }
+  }
+
+  private fun addListeners() {
+    val listeners = Listeners()
+    rangePicker!!.addOnPositiveButtonClickListener(listeners)
+    rangePicker!!.addOnDismissListener(listeners)
+  }
+
+  private fun show() {
+    rangePicker!!.show(fragmentManager, MaterialRangePickerModule.NAME)
+  }
+
+  private inner class Listeners : MaterialPickerOnPositiveButtonClickListener<Pair<Long, Long>>,
+    DialogInterface.OnDismissListener {
+    override fun onDismiss(dialog: DialogInterface) {
+      if (promiseResolved || !reactContext.hasActiveReactInstance()) return
+
+      val result = WritableNativeMap()
+      result.putString("action", RNConstants.ACTION_DISMISSED)
+      promise.resolve(result)
+      promiseResolved = true
+    }
+
+    override fun onPositiveButtonClick(selection: Pair<Long, Long>) {
+      if (promiseResolved || !reactContext.hasActiveReactInstance()) return
+
+      val result = WritableNativeMap()
+
+      result.putString("action", RNConstants.ACTION_RANGE_SET)
+      result.putDouble("startTimestamp", getStartTimestamp(selection))
+      result.putDouble("endTimestamp", getEndTimestamp(selection))
+      result.putDouble(
+        "utcOffset",
+        getStartTimestamp(selection) / 1000 / 60
+      )
```

Comment:
```suggestion
      val result = WritableNativeMap().apply {
       putString("action", RNConstants.ACTION_RANGE_SET)
       putDouble("startTimestamp", getStartTimestamp(selection))
       putDouble("endTimestamp", getEndTimestamp(selection))
       putDouble(
        "utcOffset",
        getStartTimestamp(selection) / 1000 / 60
      )
 }
```

more koltin-y. You can use `apply` in many places throughout this file to save keystrokes